patches/minecraft/net/minecraft/block/IBubbleSource.java.patch

```diff
@@ -0,0 +1,26 @@
+--- /dev/null
++++ b/net/minecraft/block/IBubbleSource.java
+@@ -0,0 +1,23 @@
++package net.minecraft.block;
++
++import net.minecraft.util.math.BlockPos;
++import net.minecraft.world.IWorldReader;
++import net.minecraft.world.World;
++import java.util.Random;
++
++
++import java.util.Random;
++
++public interface IBubbleSource {
++    default void tick(BlockState state, World worldIn, BlockPos pos, Random random) {
++        BubbleColumnBlock.func_203159_a(worldIn, pos.func_177984_a(), this.isDragging());
++    }
++
++    default int tickRate(IWorldReader worldIn) { return 20; }
++
++    default void onBlockAdded(BlockState state, World worldIn, BlockPos pos, BlockState oldState, boolean isMoving) {
++        worldIn.func_205220_G_().func_205360_a(pos, (Block) this, this.tickRate(worldIn));
++    }
++
++    boolean isDragging();
++}
```

Comment:
two things:
1. why is this in the patches folder instead of forge sources?
2. You can't introduce methods "above" the vanilla class hierarchy with the same name like your do here since the vanilla methods will be reobfed but these won't.

Reply:
1. Right. I have to change that.
2. I see. So basically drop every vanilla method in the interface? It wouldn't result in any errors if they're missing anyway. A functioning bubble source block would still need them.